src/plugins/auto/plugin.cpp

```diff
@@ -71,7 +71,7 @@ std::map<unsigned int, std::list<std::string>> MultiDeviceInferencePlugin::_prio
 
 std::map<std::string, std::string> MultiDeviceInferencePlugin::GetSupportedConfig(
     const std::map<std::string, std::string> & config, const std::string & deviceName) const {
-    std::vector<std::string> supportedConfigKeys = GetCore()->GetMetric(deviceName, METRIC_KEY(SUPPORTED_CONFIG_KEYS));
+    auto supportedConfigKeys = GetCore()->GetMetric(deviceName, METRIC_KEY(SUPPORTED_CONFIG_KEYS)).as<std::vector<std::string>>();
```

Comment:
will we remove `as<std::vector<std::string>>()` once `METRIC_KEY(SUPPORTED_CONFIG_KEYS)` is replaced with typed property?

Reply:
Yes we can do it when we add templated `get|set` methods inside internal classes